README.md

```diff
@@ -14,107 +14,102 @@ With Leiningen:
 
 ### Writing events
 
-Define a configuration.
+Events must contain a value key, which describes the message, and may also
+contain a optional `:key` key, which will be used to route particular messages to
+particular topic partitions, and can be accessed for other purposes. Events are
+converted to json before sending, which is lossy for namespaced keys.
 
 ```clojure
 (require '[pyroclast-clojure.v1.client :as client])
 
 (def config
-  {:write-api-key "<token>"
-   :topic-id "<topic-id>"
-   :format :json})
-```
+  {:pyroclast.topic/id "topic-35e284dc-8b0c-4866-b33b-8895c25ff7f6"
+   :pyroclast.topic/write-key "d8b5869d-69a9-4935-9a35-cd0f67379124"
+   :pyroclast.api/region "us-west-2"})
 
-Events must contain a value key, which describes the message, and may also
-contain a optional key key, which will be used to route particular messages to
-particular topic partitions, and can be accessed for other purposes.
+;; Send a single event asynchronously, returning a promise
+(def resp (topic-send-event! config {:value {:event-type "page-visit" :page "/home" :timestamp 1495072835000}}))
+;; Use clojure.core/deref to block on promise resolution
+@resp ;; => true
+;; Can use deref/@ inline for a synchronous operation.
+@(topic-send-event! config {:value {:event-type "page-visit" :page "/home" :timestamp 1495072835000}}) ;; => true
 
-#### Send one event synchronously
+;; Batch events together for better performance
+@(topic-send-events! config [{:value {:event-type "page-visit" :page "/home" :timestamp 1495072835000}}
+                             {:value {:event-type "page-visit" :page "/home" :timestamp 1495072835032}}]) ;; => true
 
-```clojure
-(client/send-event! config {:value {:event-type "page-visit" :page "/home" :timestamp 1495072835000}})
 ```
 
-Response: `{:created true}`
+### Reading events
 
-#### Send a batch of events synchronously
+Read events from Pyroclast Topics.
 
 ```clojure
-(client/send-events! config [{:value {:event-type "page-visit" :page "/home" :timestamp 1495072835000}}
-                             {:value {:event-type "page-visit" :page "/console" :timestamp 1495072895032}}])
-```
-
-Response: `{:created true}`
-
-#### Send one event asynchronously
+(require '[pyroclast-clojure.v1.client :as client])
 
-```clojure
-(client/send-event-async!
-  config (fn [result] (println result))
-  {:value {:event-type "page-visit" :page "store" :timestamp 1495072835000}})
-```
+(def config
+  {:pyroclast.topic/read-key "db621c74-0d8d-41ed-b4e2-d4279e8b46f8"
+   :pyroclast.topic/id "topic-35e284dc-8b0c-4866-b33b-8895c25ff7f6"
+   :pyroclast.api/region "us-west-2"})
 
-#### Send a batch of events asynchronously
+;; Subscribe to a topic, registering a new Consumer Group
+(def consumer-instance-map @(client/topic-subscribe config "my-consumer-group"))
```

Comment:
Lets rename consumer-instance-map to consumer-instance? I think it's fine to make topic-subscribe block so they don't have deref too. It's not like the consumer can do anything in the mean time?